File: packages/shared/types/patient.ts
/**
 * Shared Patient types used across mobile app and API responses.
 */

export interface Patient {
  id: string;
  age_range?: string;
  weight_range?: string;
  created_at: string;
}

export interface Medication {
  drugbank_id: string;
  drug_name: string;
  dose: string;
  frequency: string;
  start_date: string;
  indication?: string;
}

File: packages/api/models/patient.py
"""
Cascadex API — Patient Pydantic Models.
"""

from typing import Optional

from pydantic import BaseModel


class PatientCreate(BaseModel):
    age_range: Optional[str] = None
    weight_range: Optional[str] = None


class PatientProfile(BaseModel):
    id: str
    age_range: Optional[str] = None
    weight_range: Optional[str] = None
    created_at: Optional[str] = None


class MedicationAdd(BaseModel):
    drugbank_id: str
    dose: Optional[str] = None
    frequency: Optional[str] = None
    indication: Optional[str] = None


class MedicationInfo(BaseModel):
    drugbank_id: str
    name: str
    drug_class: Optional[str] = None
    dose: Optional[str] = None
    frequency: Optional[str] = None
    start_date: Optional[str] = None
